### DISCUSSION TITLE
Is it possible to preset/save green box selection

### DISCUSSION THREAD

**Discussion by dgoryeo** [YaoFANGUK/video-subtitle-extractor#163] [Q&A] [answered]
Hi,

Just came across VSE. Thanks a lot for making it available --it looks promising. This is my first experiment with PaddlePaddle based work. 

Quick question / idea: would it be possible to save the settings for the green box area selection? This would save quite a bit of time for repeated projects or videos that have same sub areas. 

Thanks

**Comment by YaoFANGUK** [comment]
Of course. I will implement it asap

**Reply by YaoFANGUK** [reply]
Done, plz use the latest source code..

**Reply by dgoryeo** [reply]
I just tried the latest source code. Here are few quick observations --hope this can help with the next version:

- It uses the CPU, though I had installed everything acording to the requirements_gpu.txt. The biggest CPU user seems to be the process for videosubfinder. I wonder if that process is not compiled or called with GPU option.

- I tried couple of videos for extracting subs. Did quite a nive job with one, but missed quite a bit in another one. The one that missed was using multi-color subs in a karaok/typewriter style. If VSE is using videosubfinder (vsf) for subtitles frame detection, there are config settings for vsf that address those multi-color typewriter subs. I can send a sample of config to you if that helps. 

-  I couldn't find the place to save the settings for the green area --I looked in the settings.ini. It probably is some where else. 

Good work and it is a promising tool. I hope it can solve the detection problem for those movies with multi-clour typewriter subs.

**Reply by YaoFANGUK** [reply]
Thank you for your input. Here are some reasons why the problems you mentioned exist:

- It's true that videosubfinder is the biggest consumer of CPU and I found you may turn on GPU acceleration of videosubfinder by setting ```use_cuda_gpu``` to ```1``` in ```<path_to_vse>/vse/backend/subfinder/settings/general.cfg```

<img src="https://s1.ax1x.com/2022/11/24/zGKD2D.png" width="500">

- Limited by the algorithm itself, which compares the difference of one frame with previous frame, this software DO NOT support detect karaok/typewriter style subs currently. It provides two modes which are called ```fast``` and ```accudate```, if you are using Windows operations system and run fast mode, VSE will use videosubfinder for subtitles frame detection. Howere, if you use OS other than Windows, whatever mode you use, VSE will extract  frames at a specific frequency which you may change -the variable name is EXTRACT_FREQUENCY- in ```backend/config.py```. it will be appreciated if you can send me a sample of config that address multi-color typerwriter subs and a sample video as well.

 - The place to save the setting for the green area is a file called ```subtitle.ini``` which is generated after your click of ```RUN```

I will try to solve the problem for multi-colour typewriter subs extraction but the next step forward is to remove subtitles after extraction:

<img src="https://s1.ax1x.com/2022/11/24/zGQrBd.png" width="500">

**Comment by dgoryeo** [comment]
Here is a config file for VSF that manages multi-color and typewriter subs. 


[VSF_Rainbow+Karaoke_MZ_V5_wcuda.zip](https://github.com/YaoFANGUK/video-subtitle-extractor/files/10087832/VSF_Rainbow%2BKaraoke_MZ_V5_wcuda.zip)


Hope this helps

**Reply by YaoFANGUK** [reply]
Thank you very much :)

**Comment by YaoFANGUK** [comment, marked as answer]
supported

**Comment by dgoryeo** [comment]
Thanks @YaoFANGUK , I just tried the latest version but I cannot work around this error:
```

python gui.py
Traceback (most recent call last):
  File "gui.py", line 8, in <module>
    import backend.main
  File "/mnt/g/paddle/video-subtitle-extractor/backend/main.py", line 25, in <module>
    import config
  File "/mnt/g/paddle/video-subtitle-extractor/backend/config.py", line 16, in <module>
    from paddle import fluid
  File "/home/mka/miniconda3/envs/videoEnv/lib/python3.8/site-packages/paddle/__init__.py", line 31, in <module>
    from .framework import monkey_patch_variable
  File "/home/mka/miniconda3/envs/videoEnv/lib/python3.8/site-packages/paddle/framework/__init__.py", line 17, in <module>
    from . import random  # noqa: F401
  File "/home/mka/miniconda3/envs/videoEnv/lib/python3.8/site-packages/paddle/framework/random.py", line 17, in <module>
    from paddle import fluid
  File "/home/mka/miniconda3/envs/videoEnv/lib/python3.8/site-packages/paddle/fluid/__init__.py", line 36, in <module>
    from . import framework
  File "/home/mka/miniconda3/envs/videoEnv/lib/python3.8/site-packages/paddle/fluid/framework.py", line 33, in <module>
    from .proto import framework_pb2, data_feed_pb2
  File "/home/mka/miniconda3/envs/videoEnv/lib/python3.8/site-packages/paddle/fluid/proto/framework_pb2.py", line 5, in <module>
    from google.protobuf.internal import builder as _builder
ImportError: cannot import name 'builder' from 'google.protobuf.internal' (/home/mka/miniconda3/envs/videoEnv/lib/python3.8/site-packages/google/protobuf/internal/__init__.py)

```
